File: Proyecto-PracticasProfesionales/backend/models/notificacion.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Boolean
from database import Base
from sqlalchemy.sql import func

class Notificacion(Base):
    __tablename__ = "notificacion"

    id_notificacion = Column(Integer, primary_key=True, index=True, autoincrement=True)
    id_usuario = Column(Integer, nullable=False)
    titulo = Column(String(150), nullable=True)
    mensaje = Column(Text, nullable=True)
    leida = Column(Boolean, default=False)
    fecha_envio = Column(DateTime, server_default=func.now())

File: Proyecto-PracticasProfesionales/src/domain/notificaciones/Notificacion.ts
export interface Notificacion {
  id_notificacion: number;
  id_usuario: number;
  titulo: string;
  mensaje: string;
  leida: boolean;
  fecha_envio: string;
}

export interface ResumenNotificaciones {
  total: number;
  no_leidas: number;
  leidas: number;
}
